### ISSUE TITLE
Select all shortcut selects all blocks instead of content in block (Microsoft Edge)

### ISSUE THREAD

**Issue by grappler** [WordPress/gutenberg#19102]
**Describe the bug**
In Microsoft Edge the select all shortcut `ctrl+A` selects all blocks instead of only selecting all of the content in a block.

This is not consistent with the behaviour in Chrome and Firefox. In other browsers only when using the shortcut again are all of the blocks selected.

**To reproduce**
1. Edit a post with multiple posts in Microsoft Edge
2. Use the select all shortcut `ctrl+A`
3. All blocks are selected

**Expected behavior**
Only the text within the paragraph block should be selected not all of the blocks.

**Environment**
- OS: Windows 10
- Browser: Edge
- Version: 42.17134.1.0
- WordPress: 5.3
- Gutenberg: 7.1

**Comment by MarcoZehe** [comment]
Not only a problem in Edge, I also reproduced this in Firefox the other day.

**Comment by grappler** [comment]
@MarcoZehe I was not able to reproduce this issue in Firefox. 

Here is a short screencast of Firefox: https://youtu.be/JrSH711-37g

**Comment by MarcoZehe** [comment]
Sorry, I am blind and cannot see the screen. So I'll take your word for it. I did reproduce it earlier this week, but when I tried just now, it worked fine. Not sure what's up with that or what condition triggers the behavior.

**Comment by talldan** [comment]
@grappler Is this something you're still experiencing with the latest version of the plugin?

**Comment by grappler** [comment]
@talldan Yes, this bug is still existent. Any reason why you would expect it to be fixed?

I did notice that now the first time running "select all" in a paragraph block works fine but after that it always selects all of the blocks.

**Comment by talldan** [comment]
@grappler What you describe is how it's intended to work. Press once to select all text, press a second time to select all blocks.

As such I'll close the issue.

**Comment by grappler** [comment]
@talldan But that is not the case in Edge. 

**Current Behaviour**
- Click in a paragraph block
  - Press once selects all **text**
- Click in another paragraph block 
  - Press once selects all **bocks**
- Click in another paragraph block 
  - Press once selects all **bocks**

**Expected Behaviour**
- Click in a paragraph block
  - Press once selects all **text**.
- Click in another paragraph block 
  - Press once selects all **text**
  - Press twice selects all **bocks**

Here is a screencast: https://youtu.be/djIfj67Qdbk

**Comment by talldan** [comment]
Thanks for the extra info @grappler, I can confirm that now that you've clarified how to reproduce the issue. 

The Issue is now reopened. I've also updated the issue description with those details from your latest comment.

**Comment by alexstine** [comment]
I was able to reproduce this very confusing bug today in Firefox (latest version).

When I tried to make a selection in a block while in edit mode, every block was selected. These are the steps I followed.
1. In select mode, navigate to a block.
2. Enter the block using Enter, now we're in edit mode.
3. Select all with CTRL+A.
4. It will likely work fine since this is the first block edited.
5. Switch back to select mode using Escape.
6. Choose another block and press Enter to go back to edit mode.
7. Press CTRL+A to select all and every block will select.
8. You can also use Shift+arrows to select text and if you reach the block border, you will select other blocks at this point as well which will effectively cancel your in-block selection.

I believe this to be a major UX bug but very much an accessibility issue. If a screen reader user is editing a text field, they would expect CTRL+A to select all in the text field, not every block on the page. The whole idea that Gutenberg should flow like a Google Doc just doesn't work due to the concept of blocks themselves. Blocks are meant to have borders, restricted interaction and separation from other blocks. I would not expect the CTRL+A or Shift+arrows to select anything outside of the currently edited block unless you were in select mode. I also tested this with the setting that prevents the cursor from going in to other blocks while in edit mode and still no change.

Thanks.

**Comment by joedolson** [comment]
I was able to duplicate this workflow problem, as well, and agree that it's a significant accessibility problem. Being able to use Ctrl+A to select the whole set of blocks is an ability that is needed, but I would expect it to be the less common scenario. It should not be difficult to select the contents of just one block.

**Comment by draganescu** [comment]
*Notes*

Investigating #30462 I found the problem here to be that:

- we handle the Primary+A shortcut in two places, once in `packages/block-editor/src/components/keyboard-shortcuts/index.js` and once in `packages/block-editor/src/components/writing-flow/index.js`.
- removing the keyboard shortcuts from the `keyboard-shortcuts` component has no effect.
- `WritingFlow` has a problem in the condition that wraps the call to `multiSelect` in that it checks for `entirelySelected.current` which is manually set to `true` therefore always defaulting to selecting all blocks.

Updating the condition to check for:

```JS
( target.isContentEditable &&
	isEntirelySelected( target ) ) ||
( ! target.isContentEditable && entirelySelected.current )
```

... fixes this problem BUT does not work on list blocks or on the quote part of quotes, although it does work if the focus is in the citation. So some blocks (paragraph, image, heading, buttons) work as expected, and some don't.

Will continue to check this.

**Comment by Mamaduka** [comment]
@talldan, can we close this issue now that https://github.com/WordPress/gutenberg/pull/31859 is merged?

I can no longer reproduce the described behavior. Single CMD +A selects text in the block, and CMD + A + A selects all blocks.

**Comment by talldan** [comment]
Sounds good 👍
